packages/cmrjs/tests/cmr-utils/test-cmr-utils.js

```diff
@@ -729,3 +732,159 @@ test(
   'metadataObjectFromCMRFile throws PreconditionFailed when ETag does not match CMR UMMG JSON metadata file',
   testMetadataObjectFromCMRFile('s3://bucket/fake.cmr.json')
 );
+
+test.serial('publish2CMR passes cmrRevisionId to publishECHO10XML2CMR', async (t) => {
+  // const cmrFileObject = { filename: 'test.cmr.xml', granuleId: 'testGranuleId' };
+  const bucket = randomString();
+  const payloadPath = path.join(__dirname, 'data', 'payload.json');
+  const rawPayload = fs.readFileSync(payloadPath, 'utf8');
+  const payload = JSON.parse(rawPayload);
+  const xmlFile = payload.input.granules[0].files[3];
+  const xmlGranuleId = payload.input.granules[0].granuleId;
+  const xmlKey = `${xmlGranuleId}.cmr.xml`;
+  const echoMetadataString = fs.readFileSync(path.join(__dirname, 'data/meta.xml'));
+  const cmrRevisionId = 12;
+  const credentials = await getCmrSettings();
+
+  await createBucket(bucket);
+  await s3PutObject({
+    Bucket: bucket,
+    Key: xmlKey,
+    Body: echoMetadataString,
+  });
+
+  const metadataObject = await cmrUtil.metadataObjectFromCMRFile(buildS3Uri(bucket, xmlKey));
+  const updatedXmlFile = { ...xmlFile, granuleId: xmlGranuleId, metadataObject };
+  const publishECHO10XML2CMRSpy = sinon.spy(() => Promise.resolve());
+  const revertPublishECHO10XML2CMRSpy = cmrUtil.__set__('publishECHO10XML2CMR', publishECHO10XML2CMRSpy);
+
+  t.teardown(() => {
+    revertPublishECHO10XML2CMRSpy();
+  });
+  await cmrUtil.publish2CMR(updatedXmlFile, credentials, cmrRevisionId);
```

Comment:
Since we're mocking `publishECHO10XML2CMR` and spying on it, do we need to go through all of the above setup for fixtures for this test? We could pass in an object only containing an echo 10 `filename` property for the first argument and literally anything for `credentials` and I think this test would still pass. The reason I ask is that if the test conditions can be simplified and yet still provide the same guarantees, it's preferable because it can make what the test is doing more obvious.

The same goes for the test of `publishUMMGJSON2CMR`.

Reply:
😱 I didn't realize we could pass anything in for credentials but taking a closer look, that makes sense now since I am mocking the ingest functions.